lib/IAMClient.js

```diff
@@ -457,31 +469,25 @@ class IAMClient {
             })
 
             .on('error', err => {
-                if (logger) {
-                    logger.debug('error receiving data', {
-                        component: 'vaultclient',
-                        method: 'IAMClient:request()',
-                        error: err.stack || err,
-                    });
-                    return callback(errors.InternalError);
-                }
-                return callback(err);
+                log.debug('error receiving data', {
+                    component: 'vaultclient',
+                    method: 'IAMClient:request()',
+                    error: err.stack || err,
```

Comment:
error must be a string, so let's set
```js
{
   error: err.message,
   errorStack: err.stack,
}

```
Can you update other references as well?